src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -96,15 +96,12 @@ public JsonPropertyInfo GetPolymorphicJsonPropertyInfo()
         /// </summary>
         public JsonConverter InitializeReEntry(Type type, JsonSerializerOptions options, string? propertyName = null)
         {
-            JsonClassInfo newClassInfo = options.GetOrAddClass(type);
-
-            // todo: check if type==newtype and skip below?
```

Comment:
Skipping the logic below would only have a minimal impact on perf, so it is not worth the check \ risk of doing that.